src/alert.cpp

```diff
@@ -244,8 +246,20 @@ bool CAlert::ProcessAlert(const std::vector<unsigned char>& alertKey, bool fThre
     return true;
 }
 
-void
-CAlert::Notify(const std::string& strMessage, bool fThread)
+// static
+void CAlert::NotifyInternal(const std::string& strMessage)
+{
+    static CAlert internalAlert;
+    internalAlert.vchMsg = std::vector<unsigned char>(strMessage.begin(), strMessage.end());
+    uint256 zero;
+    LOCK(cs_mapAlerts);
```

Comment:
You should limit the scope of the LOCK to just the next line, so cs_mapAlerts is not locked when NotifyAlertChanged / Notify is called.